src/site/content/en/web-dev-basics-one/fetch-api-error-handling/index.md

```diff
@@ -0,0 +1,234 @@
+
+### When the Fetch API throws errors
+
+This example uses a [try…catch](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/try...catch) statement to catch any errors thrown within the `try` block. For example, if the Fetch API cannot fetch the specified resource, then an error is thrown. Within a `catch` block like this, take care to provide a [meaningful](https://developer.mozilla.org/docs/Glossary/Graceful_degradation) user experience. If a spinner, a common user interface that represents some sort of progress, is shown to the user, then you could take the following actions within a `catch` block:
+
+1. Remove the spinner from the page.
+1. Provide helpful messaging that explains what went wrong, and what options the user can take.
+1. Based on the available options, present a "Try again" button to the user.
+1. Behind the scenes, send the details of the error to your error-tracking service, or to the back-end. This action logs the error so it can be diagnosed at a later stage.
+
+```js
+try {
+  const response = await fetch('https://website');  
+} catch (error) {
+  // TypeError: Failed to fetch
+  console.log('There was an error', error);
+}
+```
+
+At a later stage, while you diagnose the error that you logged, you can write a [test case](https://en.wikipedia.org/wiki/Test_case) to catch such an error before your users are aware something is wrong. Depending on the error, the test could be a unit, integration, or acceptance test.
+
+### When the network status code represents an error
+
+This code example makes a request to an HTTP testing service that always responds with the HTTP status code [429 Too Many Requests](https://developer.mozilla.org/docs/Web/HTTP/Status/429). Interestingly, the response does not reach the `catch` block. A 404 status, amongst certain other status codes, does return [a network error](https://developer.mozilla.org/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful) but instead resolves normally.
+
+To check that the HTTP status code was successful, you can use any of the following options:
+
+-  Use the [Response.ok](https://developer.mozilla.org/docs/Web/API/Response/ok) property to determine whether the status code was in the range from 200 to 299.
+-  Use the [Response.status](https://developer.mozilla.org/docs/Web/API/Response/status) property to determine whether the response was successful.
+-  Use any other metadata, such as [Response.headers](https://developer.mozilla.org/docs/Web/API/Response/headers), to assess whether the response was successful.
+
+```js
+let response;
+
+try {
+  response = await fetch('https://httpbin.org/status/429');  
+} catch (error) {
+  console.log('There was an error', error);
+}
+
+// Uses the 'optional chaining' operator
+if (response?.ok) {
+  console.log('Use the response here!');
+} else {
+  console.log('HTTP Response Code: ' + response?.status)
```

Comment:
```suggestion
  console.log(`HTTP Response Code: ${response?.status}`)
```

Reply:
Total nit, but it feels like a templated string would be cleaner here. Feel free to reject if you disagree.